app/Console/Commands/SyncAssetCounters.php

```diff
@@ -39,33 +39,39 @@ public function __construct()
     public function handle()
     {
         $start = microtime(true);
+
+        // We need the whole count of all assets in order to set up the progress bar
+        $assets_count = Asset::withTrashed()->count();
+        $bar = $this->output->createProgressBar($assets_count);
+
         $assets = Asset::withCount('checkins as checkins_count', 'checkouts as checkouts_count', 'userRequests as user_requests_count')
-            ->withTrashed()->get();
-
-        if ($assets) {
-            if ($assets->count() > 0) {
-                $bar = $this->output->createProgressBar($assets->count());
-
-                foreach ($assets as $asset) {
-                    $asset->checkin_counter = (int) $asset->checkins_count;
-                    $asset->checkout_counter = (int) $asset->checkouts_count;
-                    $asset->requests_counter = (int) $asset->user_requests_count;
-                    $asset->unsetEventDispatcher();
-                    $asset->save();
-                    $output['info'][] = 'Asset: '.$asset->id.' has '.$asset->checkin_counter.' checkins, '.$asset->checkout_counter.' checkouts, and '.$asset->requests_counter.' requests';
-                    $bar->advance();
-                }
-                $bar->finish();
-
-                foreach ($output['info'] as $key => $output_text) {
-                    $this->info($output_text);
-                }
-
-                $time_elapsed_secs = microtime(true) - $start;
-                $this->info('Sync executed in '.$time_elapsed_secs.' seconds');
+            ->withTrashed()->chunk(100, function ($assets) use ($bar, $start) {
```

Comment:
I think you don't need the `$start` here; I don't see it used anywhere in the function, only after the `chunk()` command.

Reply:
You're right - that was leftover from some earlier refactoring